lalrpop/src/lr1/build/mod.rs

```diff
@@ -336,7 +336,7 @@ impl LookaheadBuild for Nil {
 
 impl LookaheadBuild for TokenSet {
     fn epsilon_moves<'grammar>(lr: &LR<'grammar, Self>,
-                               nt: NonterminalString,
+                               nt: &NonterminalString,
```

Comment:
I definitely approve of this pattern; pass atoms by ref on the stack, clone to move into a struct

Reply:
My approach so far is that if the function ever passes the type by reference, then it accepts a reference. If it only passes the type by value, then it accepts by value.

Reply:
OK, that works. For things that are cheap to clone (e.g., Rc and Arc), I typically take a stronger rule:

- All variables on the stack are by reference
- All things in structs are by value

Reply:
(But either one is fine.)